### DISCUSSION TITLE
Selecting an autocomplete suggestion deletes everything right of the cursor

### DISCUSSION THREAD

**Discussion by twiclo** [hrsh7th/nvim-cmp#1541] [Q&A]
If I start typing to the left of a word, with no space between my cursor and the next word, the auto complete window will pop up. If I select an option it deletes everything right of the cursor until there's a space. How do I disable this?

**Comment by fitrh** [comment]
Can you share your cmp keymap config? I think you need to map your confirm key to

```lua
cmp.mapping.confirm({ behavior = cmp.ConfirmBehavior.Insert })
```

**Comment by twiclo** [comment]
```
local cmp = require('cmp')
local cmp_enabled = cmp.get_config().enabled
local cmp_config = lsp.defaults.cmp_config({
  enabled = function()
    if require('cmp.config.context').in_treesitter_capture('comment') == true
      or require('cmp.config.context').in_syntax_group('Comment')
    then
      return false
    else
      -- execute cmp's original function
      return cmp_enabled()
    end
  end
})
cmp.setup(cmp_config)
```

**Reply by fitrh** [reply]
Where is the mapping config? What is the `lsp` variable? It seems you are using some sort of preconfigured thing, if so, it is better to consult with them first

**Reply by twiclo** [reply]
Will do. I'll report back when I know more for future readers
